### PR TITLE
chore: reduce changelog size by removing text before stable code generation

### LINKED ISSUE #18210: Prevent unbounded growth in changelogs
Since the CHANGELOG.md files aren't pruned from a module we need to ensure that their growth isn't unbounded.  E.g. for NRP the changelog is currently ~1.5MB which is much larger than the combined source files.  While this is a bit of an outlier, we must ensure the files aren't simply allowed to grow.

### PR DESCRIPTION
**Pull request by tadelesh**
Close #18210 

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Comment by lirenhe** [comment]
@xboxeer, please help to review the content used in the initial changelog file. Thanks

File: sdk/resourcemanager/aad/armaad/CHANGELOG.md [hunk 1]
```diff
@@ -1,239 +1,9 @@
...
-- Function `*PrivateEndpointConnectionsClient.BeginDelete` return value(s) have been changed from `(*armruntime.Poller[PrivateEndpointConnectionsClientDeleteResponse], error)` to `(*runtime.Poller[PrivateEndpointConnectionsClientDeleteResponse], error)`
-- Function `*PrivateLinkForAzureAdClient.BeginCreate` return value(s) have been changed from `(*armruntime.Poller[PrivateLinkForAzureAdClientCreateResponse], error)` to `(*runtime.Poller[PrivateLinkForAzureAdClientCreateResponse], error)`
-- Function `ErrorDefinition.MarshalJSON` has been removed
-- Function `PrivateLinkResourceProperties.MarshalJSON` has been removed
-- Function `PrivateLinkPolicyListResult.MarshalJSON` has been removed
-- Function `PrivateEndpointConnectionListResult.MarshalJSON` has been removed
-- Function `PrivateLinkResourceListResult.MarshalJSON` has been removed
+The package of `github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/aad/armaad` is using our next generation design principles since version 1.0.0, which contains breaking changes.
```
Comment (xboxeer) [inline sdk/resourcemanager/aad/armaad/CHANGELOG.md:5]: next generation design principles link to https://azure.github.io/azure-sdk/general_introduction.html

File: sdk/resourcemanager/advisor/armadvisor/CHANGELOG.md [hunk 2]
```diff
@@ -1,329 +1,9 @@
...
 ## 1.0.0 (2022-05-17)
-### Breaking Changes
 
-- Function `OperationEntityListResult.MarshalJSON` has been removed
-- Function `ConfigurationListResult.MarshalJSON` has been removed
-- Function `ResourceRecommendationBaseListResult.MarshalJSON` has been removed
-- Function `MetadataEntityListResult.MarshalJSON` has been removed
-- Function `SuppressionContractListResult.MarshalJSON` has been removed
-- Function `MetadataEntityProperties.MarshalJSON` has been removed
+The package of `github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/advisor/armadvisor` is using our next generation design principles since version 1.0.0, which contains breaking changes.
```
Comment (xboxeer) [inline sdk/resourcemanager/advisor/armadvisor/CHANGELOG.md:5]: next generation design principles link to https://azure.github.io/azure-sdk/general_introduction.html

**Review by xboxeer — COMMENTED** [review]
next generation design principles link to https://azure.github.io/azure-sdk/general_introduction.html

**Review by xboxeer — APPROVED** [review]
looks good to me

**Pull request merged**